cpu/stm32/periph/timer.c

```diff
@@ -74,12 +163,46 @@ int timer_set_absolute(tim_t tim, int channel, unsigned int value)
         return -1;
     }
 
+    set_oneshot(tim, channel);
+
     TIM_CHAN(tim, channel) = (value & timer_config[tim].max);
-    dev(tim)->SR &= ~(TIM_SR_CC1IF << channel);
+
+#ifdef MODULE_PERIPH_TIMER_PERIODIC
+    if (dev(tim)->ARR == TIM_CHAN(tim, channel)) {
+        dev(tim)->ARR = timer_config[tim].max;
+    }
+#endif
+
+    dev(tim)->DIER |= (TIM_DIER_CC1IE << channel);
+
+    return 0;
+}
+
+#ifdef MODULE_PERIPH_TIMER_PERIODIC
+int timer_set_periodic(tim_t tim, int channel, unsigned int value, uint8_t flags)
+{
+    if (channel >= (int)TIMER_CHANNEL_NUMOF) {
+        return -1;
+    }
+
+    clear_oneshot(tim, channel);
+
+    if (flags & TIM_FLAG_RESET_ON_SET) {
+        /* setting COUNT gives us an interrupt on all channels */
+        set_ignore_irq(tim);
+        dev(tim)->CNT = 0;
+    }
```

Comment:
Have you tried if wrapping this is `irq_disable()` ... `irq_restore()` and clearing the interrupt flags from the status registers works? IMO, that would be favorable over conditionally ignoring IRQs

Reply:
Thank you, that simplifies things quite a bit.